### PR TITLE
feat: SkillsGrid

### PR DESCRIPTION
**Pull request by theogainey**
### Description

add SkillsGrid component and styles
add SkillsGrid to details page
add getSkills utility function and tests
- getSkills can accept both CSV strings and arrays of strings

### Spec

Designs: [Designs](https://www.figma.com/file/kswpPSHH2h5njVyfGDbIHk/FSA-Capstone-Project-2022)

See Story: [196](https://sparkbox.atlassian.net/browse/FSA22V1-196)

### Validation

* [x] This PR has visual elements, so it was reviewed by a designer.
* [x] This PR has code changes, and our linters still pass.
* [x] This PR affects production code, so it was browser tested (see below).
* [x] This PR has new code, so new tests were added or updated, and they pass.
* [x] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, Snyk, etc).
2. Pull down all related branches.
3. Confirm all tests pass.

### CODE REVIEW

File: styles/components/_skills-grid.scss
Comment (QuinnrElder) [inline styles/components/_skills-grid.scss:19]: LGTM! Nitpick ~ maybe think about alphabetizing the properties.

File: pages/[id].jsx
Comment (QuinnrElder) [inline pages/[id].jsx:34]: Have we thought about what the skills list should look like if there are no values? Do we want to show that column at all or use some filler text? What are your thoughts on this?? 

Screen Shot 2022-05-18 at 12 21 06 PM

File: util/getSkills.js
Comment (QuinnrElder) [inline util/getSkills.js:2]: Really like the use of SRP (single responsibility principle) here. It is easy to read and these keys can easily be updated in the future. 👍

File: util/getSkills.js
Comment (QuinnrElder) [inline util/getSkills.js:16]: Like the use of Set() ~ @dustin-jw made a good recommendation. Glad you chose to use it here!

File: pages/[id].jsx
Comment (theogainey) [reply pages/[id].jsx:34]: Earlier today I reached out to Merani about this and am waiting to hear back about her thoughts on how filtering or not filtering fits with the design. I thought I put in the PR description a "don't merge till I confirm this" but I looks like I forgot. Take a look at Legolas and you'll see an even clearer picture that you are right, we do need to think about this.

File: util/getSkills.test.js
Comment (QuinnrElder) [inline util/getSkills.test.js:26]: I love tests!! LOL

In this test are we not testing the same output as the test above? (duplicate)

File: util/getSkills.test.js
Comment (QuinnrElder) [inline util/getSkills.test.js:43]: Really like that you are testing sad paths here as well as the happy paths! Good stuff!!!

File: components/SkillsGrid.jsx
Comment (QuinnrElder) [inline components/SkillsGrid.jsx:6]: https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

This is good! This was shared in Dev the other day and I thought it was thought-provoking.

File: components/SkillsGrid.jsx
Comment (QuinnrElder) [inline components/SkillsGrid.jsx:5]: https://www.youtube.com/watch?v=ULdkpU51hTQ

This was shared with me by Kasey a month or so ago. From an accessibility standpoint, we might be able to switch out the parent div with a section kinda like:

```
<section aria-labelledby="Skills-Header" className="cmp-skills-grid"> 
            <h2 id="Skills-Header" className="cmp-skills-grid__heading">Skills</h2> 
```

Comment (theogainey) [reply components/SkillsGrid.jsx:5]: I listened to both with voice over and yeah this is the better way to go.

File: pages/[id].jsx
Comment (theogainey) [reply pages/[id].jsx:26]: After talking to Merani the answer is to not show columns that have nothing in them. I made this happen by adding a filter in getSkills

Comment (QuinnrElder) [reply pages/[id].jsx:26]: Looks great!

Screen Shot 2022-05-19 at 8 28 25 AM

**Review by QuinnrElder — APPROVED** [review]
The code looks great, Theo! I was impressed with your approach to SRP and your testing of sad paths! Keep up the excellent work. Make sure that you have a designer check that last box in your PR checklist before merging; if you have not already.

**Comment by theogainey** [comment]
For what Quinn said about the design. I showed Merani my solution and she said it looked good.

**Pull request merged**
